File: src/net/auoeke/cin/lexer/lexeme/BooleanLexeme.java
package net.auoeke.cin.lexer.lexeme;

import net.auoeke.cin.lexer.error.SyntaxError;

public final class BooleanLexeme extends Lexeme {
    public final boolean value;

    public BooleanLexeme(int line, int column, boolean value, SyntaxError error) {
        super(line, column, error);

        this.value = value;
    }

    public BooleanLexeme(int line, int column, boolean value) {
        this(line, column, value, null);
    }

    @Override public Type type() {
        return Type.BOOLEAN;
    }

    @Override public String toString() {
        return Boolean.toString(this.value);
    }
}

File: src/net/auoeke/cin/lexer/lexeme/BooleanLexeme.kt
package net.auoeke.cin.lexer.lexeme

import net.auoeke.cin.lexer.SyntaxError
import net.auoeke.extensions.string

class BooleanLexeme(line: Int, column: Int, val value: Boolean, error: SyntaxError? = null) : Lexeme(line, column, error) {
    override val type: Type get() = Type.BOOLEAN

    override fun toString(): String = value.string
}
